pkg/configprovider/opampprovider/provider.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package opampprovider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/SumoLogic/sumologic-otel-collector/pkg/configprovider/globprovider"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	SchemeName = "opamp"
+)
+
+// ConfigFragment is part of a larger opamp configuration structure.
+// To avoid creating a dependency on other packages, we only specify what
+// is needed for the OpAmp provider to work.
+type ConfigFragment struct {
+	Extensions struct {
+		OpAmp struct {
+			RemoteConfigurationDirectory string `yaml:"remote_configuration_directory"`
+		} `yaml:"opamp"`
+	} `yaml:"extensions"`
+}
+
+func (c ConfigFragment) ConfigDir() string {
+	return c.Extensions.OpAmp.RemoteConfigurationDirectory
+}
+
+func (c ConfigFragment) Validate() error {
+	if c.ConfigDir() == "" {
+		return errors.New("remote_configuration_dir missing")
```

Comment:
@echlebek how about "opamp extension remote_configuration_directory must be specified"? More words and the correct field name.